frontend/server/src/main/java/com/amazonaws/ml/mms/util/codec/MessageCodec.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.amazonaws.ml.mms.util.codec;
+
+import com.amazonaws.ml.mms.util.messages.BaseModelRequest;
+import com.amazonaws.ml.mms.util.messages.ModelInferenceRequest;
+import com.amazonaws.ml.mms.util.messages.ModelInputs;
+import com.amazonaws.ml.mms.util.messages.ModelLoadModelRequest;
+import com.amazonaws.ml.mms.util.messages.ModelWorkerResponse;
+import com.amazonaws.ml.mms.util.messages.Predictions;
+import com.amazonaws.ml.mms.util.messages.RequestBatch;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.ByteToMessageCodec;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MessageCodec extends ByteToMessageCodec<BaseModelRequest> {
+
+    private void encodeRequestBatch(RequestBatch req, ByteBuf out) {
+        out.writeInt(req.getRequestId().length());
+        out.writeBytes(req.getRequestId().getBytes(Charset.forName("UTF-8")));
+
+        if (req.getContentType() != null) {
+            out.writeInt(req.getContentType().length());
+            out.writeBytes(req.getContentType().getBytes(Charset.forName("UTF-8")));
+        } else {
+            out.writeInt(0); // Length 0
+        }
+
+        out.writeInt(-1); // Start of List
+        for (ModelInputs input : req.getModelInputs()) {
+            encodeModelInputs(input, out);
+        }
+        out.writeInt(-2); // End of List
+    }
+
+    private void encodeModelInputs(ModelInputs modelInputs, ByteBuf out) {
+        out.writeInt(modelInputs.getName().length());
+        out.writeBytes(modelInputs.getName().getBytes(Charset.forName("UTF-8")));
+
+        if (modelInputs.getContentType() != null) {
+            out.writeInt(modelInputs.getContentType().length());
+            out.writeBytes(modelInputs.getContentType().getBytes(Charset.forName("UTF-8")));
+        } else {
+            out.writeInt(0); // Length 0
+        }
+
+        out.writeInt(modelInputs.getValue().length);
+        out.writeBytes(modelInputs.getValue());
+    }
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, BaseModelRequest msg, ByteBuf out) {
+        if (msg instanceof ModelLoadModelRequest) {
+            out.writeDouble(1.0); // SOM
```

Comment:
1 byte should be enough, use int if you really what use version

Reply:
Will move to int.